### ISSUE TITLE
AWS: "The bucket you are attempting to access must be addressed using the specified endpoint"

### ISSUE THREAD

**Issue by ptolemybarnes** [makersacademy/slack-overflow#122]
@aaronkendall and I have been reviewing the instagram challenge. We've encountered a problem with the aws domain name that Paperclip is using.

Our view looks like this:
```ruby
=image_tag @picture.image.url
```

We're expecting this to render an image element with a src URL pointing to an image stored on the AWS service. This seems to work. However, it returns an URL with a domain name formatted in a way that seems not to work. The example looks like this:
```html
<img alt="Eagle1" src="http://s3.amazonaws.com/literallyanything/pictures/images/000/000/005/original/Eagle1.jpg?1445276283">
```
This doesn't load an image. But if you visit that URL, you get a page that looks like this:
```html
<Error>
   <Code>PermanentRedirect</Code>
   <Message>The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.</Message>
   <Bucket>literallyanything</Bucket>
   <Endpoint>literallyanything.s3.amazonaws.com</Endpoint>
   <RequestId>095C16477911086E</RequestId>      <HostId>TBfG15WSbA00/lJOCce78q29XHpmjqGBbT2dPJJ6EKzQKyBkvkVk78y7MSgjZ10bf+HbeISmTWY=</HostId>
</Error>
```
Usefully, this page tells us the domain name we **should** be using, `literallyanything.s3.amazonaws.com`. Visiting that, `http://literallyanything.s3.amazonaws.com/pictures/images/000/000/005/original/Eagle1.jpg?1445276283` we see our actual image. ^_^

In short, 
Broken: `http://s3.amazonaws.com/literallyanything`
Working: `http://literallyanything.s3.amazonaws.com`

So our question is: how can we tell paperclip to use the correct endpoint domain name?
@UsmanJ has also had this problem.

**Comment by mechacoach** [comment]
A word to the wise - format overflow questions like this:

1. What you're trying to do
2. The code to do it in a [GitHub-flavoured Markdown](https://help.github.com/articles/github-flavored-markdown/) code block
3. The error you're getting

That will help a casual browser to quickly point you in the right direction.

**Comment by tealpaintedduck** [comment]
does specifying url in Paperclip config work? Mine looks like this:
```
config.paperclip_defaults = {
    :storage => :s3,
    :s3_credentials => {
      :url => 'literallyanything.s3-us-west-2.amazonaws.com',
      :bucket => ENV["S3_BUCKET_NAME"],
      :access_key_id => ENV["AWS_ACCESS_KEY_ID"],
      :secret_access_key => ENV["AWS_SECRET_ACCESS_KEY"],
      :path =>'/:class/:attachment/:id_partition/:style/:filename',
      :s3_host_name => 's3-us-west-2.amazonaws.com'
    }
  }
```

**Comment by aaronkendall** [comment]
This worked nicely, cheers Sarah!

**Comment by UsmanJ** [comment]
I’ll try it a little later.  :)


Usman Jamil

M: +44 7415121250
E: usman.jamil@live.co.uk

**Comment by nikeshashar** [comment]
@UsmanJ I just deleted your number off Github. You can thank me later

**Comment by ptolemybarnes** [comment]
Thanks @tealpaintedduck

**Comment by UsmanJ** [comment]
WORKS!! WOOHOOO!!! :) Thanks @tealpaintedduck
